src/url.js

```diff
@@ -43,6 +43,33 @@ export function parseUrl(url) {
   return info;
 }
 
+/**
+ * Appends a query string field and value to a url. `key` and `value`
+ * will be ran through `encodeURIComponent` before appending.
+ * @param {string} url
+ * @param {string} key
+ * @param {string} value
+ * @return {string}
+ */
+export function addParamToUrl(url, key, value) {
+  const urlObj = parseUrl(url);
```

Comment:
I think that's certainly possible. What would the implementation look like? (something like, if theres no `?` or `#` just append the string, else fall back to the `parseUrl` code?)